scripts/run_lighthouse_tests.py

```diff
@@ -49,12 +60,125 @@ def run_lighthouse_checks():
+        'tar', '-xvzf', zlib_outfile_name, '-C', common.OPPIA_TOOLS_DIR
+    ])
+    zlib_path = os.path.join(common.OPPIA_TOOLS_DIR, 'zlib-1.2.11')
+    os.remove(zlib_outfile_name)
+
+    config_filepath = os.path.join(common.CURR_DIR, 'nginx.conf')
+    configure_command = [
+        './configure', '--prefix=' + NGINX_INSTALL_PATH, '--build=Ubuntu',
+        '--without-http_rewrite_module', '--with-zlib=' + zlib_path,
+        '--with-http_gzip_static_module', '--conf-path=' + config_filepath
+    ]
+    with common.CD(NGINX_DOWNLOAD_PATH):
+        subprocess.check_call(configure_command)
+        subprocess.check_call(['make'])
+        subprocess.check_call(['make', 'install'])
+        python_utils.PRINT('Installation successful')
+
+
+def start_proxy_server():
+    """Start the nginx proxy server."""
+    python_utils.PRINT('Starting proxy server...')
+    start_server_command = [NGINX_BINARY]
+    proxy_process = subprocess.Popen(start_server_command)
+    RUNNING_PROCESSES.append(proxy_process)
+
+
+def run_lighthouse_checks_with_compression():
+    """Run lighthouse checks with compression enabled."""
+    python_utils.PRINT('Checking if nginx is installed...')
+    if not os.path.exists(NGINX_INSTALL_PATH):
+        download_and_install_nginx()
+
+    constants_env_variable = '"DEV_MODE": false'
+    for line in fileinput.input(
+            files=[os.path.join('assets', 'constants.ts')], inplace=True):
+        # Inside this loop the STDOUT will be redirected to the file,
+        # constants.ts. The end='' is needed to avoid double line breaks.
+        python_utils.PRINT(
+            re.sub(
+                r'"DEV_MODE": .*', constants_env_variable, line), end='')
+
+    start_google_app_engine_server()
+    common.wait_for_port_to_be_open(APP_ENGINE_PORT)
+    start_proxy_server()
+    common.wait_for_port_to_be_open(NGINX_PORT)
     run_lighthouse_checks()
-    delete_reports()
+
+
+def cleanup():
+    """Kill the running subprocesses and server fired in this program."""
+    dev_appserver_path = '%s/dev_appserver.py' % common.GOOGLE_APP_ENGINE_HOME
+    processes_to_kill = [
+        '.*%s.*' % re.escape(dev_appserver_path),
+        '.*%s.*' % re.escape(NGINX_INSTALL_PATH)
+    ]
+
+    for p in RUNNING_PROCESSES:
+        p.kill()
+
+    for p in processes_to_kill:
+        common.kill_processes_based_on_regex(p)
+
+
+def main(args=None):
+    """Runs lighthouse checks and deletes reports."""
+    parsed_args = _PARSER.parse_args(args=args)
+
+    # This is to be used by the CI to only install nginx.
+    if parsed_args.only_install_nginx:
+        download_and_install_nginx()
+        return
+
+    atexit.register(cleanup)
+    build.main(args=['--prod_env'])
+
+    if parsed_args.disable_compression:
+        # Setting this so that it will be compatible with the config file.
+        start_google_app_engine_server(port=NGINX_PORT)
+        common.wait_for_port_to_be_open(APP_ENGINE_PORT)
+        run_lighthouse_checks()
```

Comment:
I would remove this part, I think we should enforce running this with compression.